run/SU2_preCICE_CHT.py

```diff
@@ -26,240 +26,247 @@
+		else:
+			print('ERROR : You are trying to launch a computation without initializing MPI but the wrapper has been built in parallel. Please add the --parallel option in order to initialize MPI for the wrapper.')
+		return
+
+
+	# Configure preCICE:
+	size = comm.Get_size()
+	try:
+		participant = precice.Participant(options.precice_name, options.precice_config, rank, size)
+	except:
+		print("There was an error configuring preCICE")
+		return
+
+	mesh_name = options.precice_mesh
+
+	# Check preCICE + SU2 dimensions
+	if options.nDim != participant.get_mesh_dimensions(mesh_name):
+		print("SU2 and preCICE dimensions are not the same! Exiting")
+		return
+
+	CHTMarkerID = None
+	CHTMarker = 'interface' # Name of CHT marker to couple
+
+	# Get all the tags with the CHT option
+	CHTMarkerList =  SU2Driver.GetAllCHTMarkersTag()
+
+	# Get all the markers defined on this rank and their associated indices.
+	allMarkerIDs = SU2Driver.GetAllBoundaryMarkers() # Returns all markers defined on this rank
+
+	#Check if the specified marker has a CHT option and if it exists on this rank.
+	if CHTMarker in CHTMarkerList and CHTMarker in allMarkerIDs.keys():
+		CHTMarkerID = allMarkerIDs[CHTMarker] # So: if CHTMarkerID != None, then it exists on this rank
+
+	# Number of vertices on the specified marker (per rank)
+	nVertex_CHTMarker = 0         #total number of vertices (physical + halo) on this rank
+	nVertex_CHTMarker_HALO = 0    #number of halo vertices
+	nVertex_CHTMarker_PHYS = 0    #number of physical vertices
+	iVertices_CHTMarker_PHYS = [] #indices of vertices this rank is working on
+	# Note: Datatypes must be primitive as input to SU2 wrapper code, not numpy.int8, numpy.int64, etc.. So a list is used
+
+	# If the CHT marker is defined on this rank:
+	if CHTMarkerID != None:
+		nVertex_CHTMarker = SU2Driver.GetNumberVertices(CHTMarkerID) #Total number of vertices on the marker
+		nVertex_CHTMarker_HALO = SU2Driver.GetNumberHaloVertices(CHTMarkerID)
+		nVertex_CHTMarker_PHYS = nVertex_CHTMarker - nVertex_CHTMarker_HALO
+
+		# Obtain indices of all vertices that are being worked on on this rank
+		for iVertex in range(nVertex_CHTMarker):
+			if not SU2Driver.IsAHaloNode(CHTMarkerID, iVertex):
+				iVertices_CHTMarker_PHYS.append(int(iVertex))
+
+	# Get coords of vertices
+	coords = numpy.zeros((nVertex_CHTMarker_PHYS, options.nDim))
+	for i, iVertex in enumerate(iVertices_CHTMarker_PHYS):
+		coord_passive = SU2Driver.GetInitialMeshCoord(CHTMarkerID, iVertex)
+		for iDim in range(options.nDim):
+			coords[i, iDim] = coord_passive[iDim]
+
+	# Set mesh vertices in preCICE:
+	try:
+		vertex_ids = participant.set_mesh_vertices(mesh_name, coords)
+	except:
+		print("Could not set mesh vertices for preCICE. Was a (known) mesh specified in the options?")
+		return
+
+	# Get read and write data IDs
+	precice_read = "Temperature"
+	precice_write = "Heat-Flux"
+	GetFxn = lambda *args: -1*SU2Driver.GetVertexNormalHeatFlux(*args)
```

Comment:
Does this make an assumption on the direction of the heat flux that we should document somewhere? I am not sure if we even document this for other adapters at the moment, but we should.